src/player.rs

```diff
@@ -107,6 +108,40 @@ fn try_pickup_item(ecs: &mut World) {
   }
 }
 
+fn try_open_door(ecs: &mut World) {
+  let mut positions = ecs.write_storage::<Position>();
+  let mut players = ecs.write_storage::<Player>();
+  let mut viewsheds = ecs.write_storage::<Viewshed>();
+  let player_entity = ecs.fetch::<Entity>();
+  let mut dungeon = ecs.fetch_mut::<Dungeon>();
+  let dungeon_levels = ecs.read_storage::<DungeonLevel>();
+  let player_level = dungeon_levels.get(*player_entity).unwrap();
+  let map = dungeon.get_map(player_level.level).unwrap();
+  let mut gamelog = ecs.fetch_mut::<GameLog>();
+
+  for (_player, pos) in (&mut players, &mut positions).join() {
+    let mut door_opened = false;
+    for x in (pos.x - 1)..(pos.x + 2) {
+      for y in (pos.y - 1)..(pos.y + 2) {
```

Comment:
can be 
```rust
    for x in (pos.x - 1)..=(pos.x + 1) {
      for y in (pos.y - 1)..=(pos.y + 1) {
```

Reply:
what if there are two doors within this range, for example at the corner of a room?

Reply:
It will open both of them (or all of them if you're surrounded by doors haha).  

Presuming that isn't intended behaviour, I was thinking maybe it needs to keep track of the direction of your last (attempted) move and only search for doors in that direction when the button is pressed?

Reply:
yeah, I was thinking of implementing a pattern like 
- move action does the default action for any given tile that you are moving against
- act menu presents a list of all available actions that can be taken on a given tile.

I think this is good for now as the code is isolated and we can continue to refactor later.